backend/authBackend.js

```diff
@@ -1,65 +1,54 @@
+import { getCurrentUser } from '@reshuffle/server-function';
 
-// used to validate incoming Google profiles from client
-import { OAuth2Client } from 'google-auth-library';
+const { VALID_HOSTED_DOMAINS } = process.env;
+const domains = VALID_HOSTED_DOMAINS.split(' ');
 
-// we depend on both OAUTH_CLIENT_ID and a SECURE
-// HMAC which is used to sign our tokens
-//
-// THE HMAC CANNOT BE LEAKED!!!!!!!!!!!!!!
-const {
-  REACT_APP_OAUTH_CLIENT_ID,
-  JWT_HMAC_KEY,
-  REACT_APP_VALID_HOSTED_DOMAIN,
-} = process.env;
+/**
+* validates if email is the right VALID_HOSTED_DOMAIN
+* setting getCurrentUser to false since I'm calling this function in the front end and I don't want
+* want the CRA to crash for not being logged in if user hits this function.
+/** @expose */
+export async function checkIfValidDomain() {
+  const profile = getCurrentUser(false);
 
-const client = new OAuth2Client(REACT_APP_OAUTH_CLIENT_ID);
+  if (profile === undefined) {
+    return false;
+  }
 
-/**
- * Makes the necessary requests to Google auth servers
- * to verify that the client user is actually who they
- * say they are.
- *
- * @param { string } token - idToken returned by clientside auth
- */
-async function verifyGoogleUser(token) {
-  const ticket = await client.verifyIdToken({
-    idToken: token,
-    audience: REACT_APP_OAUTH_CLIENT_ID,
-  });
-  return ticket.getPayload();
+  const emails = profile.emails.map((email) => {
+    return email.value
+  })
+
+  for (let i = 0; i < emails.length; i++) {
+    const email = emails[i]
+
+    for (let j = 0; j < domains.length; j++) {
+      const validDomain = domains[j]
+
+      if (email.includes(validDomain)) {
+        return true;
+      }
+    }
+  }
+
+  return false;
 }
 
+
 /**
- * Authenticates a Google user whose token was returned
- * from the client-side Google authentication flow. Assuming
- * the user is valid, a side-wide JWT will be returned for
- * future identification.
  *
- * @param { string } googleToken - googleToken returned by clientside auth
- *
- * @return { string } - site-wide JWT that allows future access
+ * @return {boolean} - returns if user is VALID_HOSTED_DOMAIN matches their email
  */
-/* @expose */
-export async function authenticateUser(googleToken) {
-  const payload = await verifyGoogleUser(googleToken);
-  const { hd } = payload;
-  if (hd === REACT_APP_VALID_HOSTED_DOMAIN) {
-    // store some basic (unreliable) profile information in the token
-    return jwt.sign({
-      name: payload.name,
-      email: payload.email,
-    }, JWT_HMAC_KEY);
-  } else {
-    // TODO: Format error to meet specification defined in contentBackend
-    throw new Error(`User cannot be authenticated, ${hd} is not a valid host domain`);
+export async function validateUser() {
+  const profile = getCurrentUser(true);
+
+  const email = profile.emails[0].value;
+
+  if (!checkIfValidDomain()) {
+    throw new Error(`User cannot be authenticated, ${email} is not a valid host domain.`)
```

Comment:
Error message is somewhat confusing, AFAICT you actually tested _all_ elements of `email`.  Consider
```suggestion
    throw new Error(`User cannot be authenticated, no valid host domain for ${emails.join(', ')}.`);
```
As you can see, it would be _easier_ to throw a good error message from `checkIfValidDomain`.

Reply:
`validateUser` doesn't throw the error anymore that job was given to the helper function `checkIfValidDomain`